proteuslib/util/initialization.py

```diff
@@ -0,0 +1,86 @@
+###############################################################################
+# ProteusLib Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/nawi-hub/proteuslib/"
+#
+###############################################################################
+"""
+This module contains utility functions for initialization of ProteusLib models.
+"""
+
+
+__author__ = "Adam Atia"
+
+from pyomo.environ import check_optimal_termination
+from idaes.core.util.model_statistics import degrees_of_freedom
+
+
+def check_solve(results, checkpoint=None, logger=None, fail_flag=False):
```

Comment:
I won't insist on it here (especially seeing as they will probably migrate to IDAES fairly soon), but it would be good to have unit tests for these methods.

Reply:
@andrewlee94 OK, I added unit tests for those methods despite the potential migration.